docs/references/kontena-yml-variables.md

```diff
@@ -0,0 +1,395 @@
+    type: string  # type (string, integer, boolean, uri, enum)
+    min_length: 8 # require at least 8 characters
+    from: # where to obtain a value for this variable
+      vault: wp-mysql-root # try to get a value from the vault on kontena master
+      env: MYSQL_ROOT # first try local env variable 
+      prompt: Enter a root password for MySQL or leave empty to auto generate # then ask for manual input
+      random_string: 16 # still no value, auto generate a random string
+    to:
+      vault: wp-mysql-root # send this value to the vault on kontena master
+  mysql_version:
+    type: enum # a list of predefined options
+    default: latest # default value
+    options:
+      - 5.5
+      - latest
+  use_mariadb:
+    type: boolean
+    from:
+      prompt: Use MariaDB?
+  image_mysql:
+    skip_if: use_mariadb # process this only if "use_mariadb" is falsey
+    type: string
+    value: mysql # set a fixed value
+    to:
+      env: db_image # place the value into local env variable "db_image"
+  image_mariadb:
+    only_if: use_mariadb # process this only if "use_mariadb" if truthy
+    type: string
+    value: mariadb
+    to:
+      env: db_image # place the value into local env variable "db_image"
+
+services:
+  mysql:
+    image: ${db_image}:${mysql_version} # use the variables
+    stateful: true
+    secrets:
+      - secret: wp-mysql-root # expose MYSQL_ROOT_PASSWORD to the container from vault key wp-mysql-root
+        name: MYSQL_ROOT_PASSWORD
+        type: env
+```
+
+End result: A MySQL stack where you can select to use MariaDB, select a version and place the root password into vault on Kontena Master.
+
+The conditional logic can also be used inside services definition:
+
+```
+services:
+  mysql:
+    skip_if: use_external_database
+```
+
+
+## Anatomy of a variable definition
+
+### The variable name
+
+```
+variables:
+  this_is_the_name:
+```
+
+By default it gets set to local env with the same name, making it possible to use the value later in the YAML:
+
+```
+  environment:
+    - ENV_VAR=${this_is_the_name}
+```
+
+It is also used when referencing this variable in conditional statements.
+
+### From
+
+Define where a value for the variable is obtained from. A list of resolvers and their options can be found further in this documentation.
+
+You can define multiple sources, for example:
+
+```
+  from:
+    env: MYSQL_USER # First try local environment variable
+    file: # Then try to read from a file
+      path: /tmp/mysql_user.txt
+      ignore_errors: true
+    prompt: Enter MySQL username # finally ask for a value if still not found
+```
+
+### To
+
+Define what to do with the value. The default behavior is to set it to local environment using the variable name. You can also use another environment variable name or write the value to the Vault on Kontena Master.
+
+```
+  to:
+    env: MYSQL_USER # set to local env
+    vault: foo-mysql-user # also send to vault
+```
+
+### Only_if
```

Comment:
These look a bit odd, maybe it would be better to just use fixed-width for these titles

#### `only_if`